### PR TITLE
Jetpack Social: Add icon download for social media icons that aren't bundled

### PR DESCRIPTION
**Pull request by wargcm**
See: #20784

## Description

Adds an icon download to the no connections view and matches the `PublicizeService` objects that are cached for the blog.

## Screenshots

| Light | Dark |
|------|------|
| Screenshot Light | Screenshot Dark |

## Testing

To test:
- Launch Jetpack and login
- Select a site which has no social connections but has services available
- Ensure the `PublicizeService` objects are cached (this can be done by going to the social screen)
- Create a blog post or edit one
- On the editor, open the menu (`...`) in the top right
- Tap on `Post Settings`
- **Verify** the icons match the services
- Tap back
- Disable your internet
- Open the `Post Settings` again
- **Verify** the placeholder is used for icons on services that we don't bundle an icon for

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Jetpack/Social/JetpackSocialNoConnectionView.swift [hunk 1]
```diff
@@ -59,20 +59,81 @@ extension JetpackSocialNoConnectionView {
...
          onConnectTap: (() -> Void)? = nil,
          onNotNowTap: (() -> Void)? = nil) {
         self.padding = padding
         self.hideNotNow = hideNotNow
         self.onConnectTap = onConnectTap
         self.onNotNowTap = onNotNowTap
+        updateIcons(services)
+    }
+
+    enum JetpackSocialService: String {
```
Comment (dvdchr) [inline WordPress/Classes/ViewRelated/Jetpack/Social/JetpackSocialNoConnectionView.swift:81]: We'll need to add Mastodon and Instagram here (along with the icons from Figma), but I'll let you decide whether to add them in this PR or do it separately.

File: WordPress/Classes/ViewRelated/Jetpack/Social/JetpackSocialNoConnectionView.swift [hunk 2]
```diff
@@ -59,20 +59,81 @@ extension JetpackSocialNoConnectionView {
...
+        case linkedin
+        case unknown
+
+        var image: UIImage? {
+            switch self {
+            case .facebook:
+                return UIImage(named: "icon-facebook")
+            case .twitter:
+                return UIImage(named: "icon-twitter")
+            case .tumblr:
```
Comment (dvdchr) [inline WordPress/Classes/ViewRelated/Jetpack/Social/JetpackSocialNoConnectionView.swift:96]: Just for Tumblr, we'll need to differentiate the icon between light and dark schemes. Same with my previous comment, I'll let you decide whether to add them here or do it later.

File: WordPress/Classes/ViewRelated/Post/PostSettingsViewController+JetpackSocial.swift [hunk 3]
```diff
@@ -25,4 +32,10 @@ extension PostSettingsViewController {
         return viewController.view
     }
 
+    private func availableServices() -> [PublicizeService] {
+        let context = apost.managedObjectContext ?? ContextManager.shared.mainContext
+        let services = try? PublicizeService.allPublicizeServices(in: context)
+        return  services ?? []
```
Comment (dvdchr) [inline WordPress/Classes/ViewRelated/Post/PostSettingsViewController+JetpackSocial.swift]: Nit: there's an extra space here.
```suggestion
        return services ?? []
```

**Review by dvdchr — APPROVED** [review]
Left some comments since there are updates to the design, along with one tiny nitpick.

Overall looks good to me!

File: WordPress/Classes/ViewRelated/Jetpack/Social/JetpackSocialNoConnectionView.swift (hunk 1 above)
Comment (wargcm) [reply WordPress/Classes/ViewRelated/Jetpack/Social/JetpackSocialNoConnectionView.swift:81]: Yep, I can add those here! I'm going to wait for this thread to resolve though before making those changes: p1687816953471379-slack-C05362VSXFU

File: WordPress/Classes/ViewRelated/Jetpack/Social/JetpackSocialNoConnectionView.swift (hunk 2 above)
Comment (wargcm) [reply WordPress/Classes/ViewRelated/Jetpack/Social/JetpackSocialNoConnectionView.swift:96]: Is there a reason we need to separate the light and dark versions? It's currently grouped with a light and dark icon and automatically uses the correct one:

Screenshot 2023-06-26 at 6 22 03 PM

I did need to update the dark icon asset though. I did that in https://github.com/wordpress-mobile/WordPress-iOS/pull/20823/commits/7ebf53681da209855c3f3c099f2af6c79a8c516c.

File: WordPress/Classes/ViewRelated/Post/PostSettingsViewController+JetpackSocial.swift (hunk 3 above)
Comment (wargcm) [reply WordPress/Classes/ViewRelated/Post/PostSettingsViewController+JetpackSocial.swift]: Fixed in https://github.com/wordpress-mobile/WordPress-iOS/pull/20823/commits/986811cde481460b1ba3c7189e18b63875737c4d, thanks for the catch!

File: WordPress/Classes/ViewRelated/Jetpack/Social/JetpackSocialNoConnectionView.swift (hunk 2 above)
Comment (dvdchr) [reply WordPress/Classes/ViewRelated/Jetpack/Social/JetpackSocialNoConnectionView.swift:96]: > Is there a reason we need to separate the light and dark versions? It's currently grouped with a light and dark icon and automatically uses the correct one

Yes, separating light and dark from the assets works for me. 👍🏼

File: WordPress/Classes/ViewRelated/Jetpack/Social/JetpackSocialNoConnectionView.swift (hunk 1 above)
Comment (wargcm) [reply WordPress/Classes/ViewRelated/Jetpack/Social/JetpackSocialNoConnectionView.swift:81]: I've added the social icons for Instagram and Mastodon in: https://github.com/wordpress-mobile/WordPress-iOS/pull/20823/commits/6be84eb9437c4422caf6588d35c95980407a3ed0

Additionally, I've updated Tumblr's dark icon with https://github.com/wordpress-mobile/WordPress-iOS/pull/20823/commits/007df21d17df580e178043ec3e9eef505d1ef1be after this conversation: p1687819305411249-slack-C05362VSXFU

Here are a couple screenshots (Note: I replaced Facebook's icon with Instagram here for testing):

| Light | Dark |
|------|------|
| Screenshot 2023-06-27 at 2 22 40 PM | Screenshot 2023-06-27 at 2 22 50 PM |

**Pull request merged**